packages/internal-test-env/index.ts

```diff
@@ -78,100 +77,97 @@ export interface TestingEnvironmentBase {
   // Common Envs
-  E2E_TEST_ENVIRONMENT: AvailableEnvironment;
+  E2E_TEST_ENVIRONMENT: AvailableEnvironments;
   E2E_TEST_IDP: string;
 }
 
 export function setupEnv() {
-  // If we're in CI, the environment is already configured.
-  if (process.env.CI) {
-    return;
-  }
-
-  if (envLoaded) {
+  // If we're in CI, the environment is already configured. Otherwise, it is
+  // loaded once.
+  if (process.env.CI || envLoaded) {
     return;
   }
-
   const envPath = join(process.cwd(), "e2e/env/.env.local");
 
-  // Otherwise load dotenv configuration
+  // Load dotenv configuration
   config({
     path: envPath,
   });
 
   if (!process.env.E2E_TEST_ENVIRONMENT) {
     console.error(
-      `We didn't find the given environment variable E2E_TEST_ENVIRONMENT,
-tried looking in the following directory for \`.env.local \`: 
-${envPath}`
+      `We didn't find the given environment variable E2E_TEST_ENVIRONMENT, tried looking in the following directory for '.env.local': ${envPath}`
     );
   }
-
-  // Creating feature flag object if there are feature flags
-  Object.keys(process.env)
-    .filter((envVar) => envVar.startsWith("E2E_TEST_FEATURE_"))
-    .forEach((key) => (featuredFlags[key] = process.env[key]));
-  // Marking env as loaded
   envLoaded = true;
 }
 
-function getTestingEnvironment(
-  environment: unknown
-): asserts environment is EnvVariables {
-  // Populate your process.env from your .env file of choice
+function getBaseTestingEnvironment<T extends LibraryVariables>(
+  libVars?: T
+): T extends NodeVariables
+  ? TestingEnvironmentNode
+  : TestingEnvironmentBrowser {
+  setupEnv();
+
+  // Load and validate target environment name.
+  const targetEnvName = process.env.E2E_TEST_ENVIRONMENT;
+  if (!availableEnvironments.includes(targetEnvName as AvailableEnvironments)) {
+    throw new Error(`Unknown environment: [${targetEnvName}]`);
+  }
 
-  // TODO: Replace these inline validations and checks with envalid or env-var
-  if (
-    !availableEnvironment.includes(
-      (environment as EnvVariables).E2E_TEST_ENVIRONMENT as AvailableEnvironment
-    )
-  ) {
+  // Load and validate target OpenID Provider.
+  const targetOp = process.env.E2E_TEST_IDP;
+  if (typeof targetOp !== "string" || !isValidUrl(targetOp)) {
     throw new Error(
-      `Unknown environment: [${
-        (environment as EnvVariables).E2E_TEST_ENVIRONMENT
-      }]`
+      `The environment variable E2E_TEST_IDP is not a valid URL: found ${targetOp}.`
     );
   }
 
-  if (typeof (environment as EnvVariables).E2E_TEST_IDP !== "string") {
-    throw new Error("The environment variable E2E_TEST_IDP is undefined.");
-  }
-}
+  // Creating feature flag object if there are feature flags
+  const features = Object.keys(process.env)
+    .filter((envVar) => envVar.startsWith(ENV_VAR_PREFIX))
+    .reduce(
+      (featureFlags, envVar) => ({
+        ...featureFlags,
+        [`${envVar}`]: process.env[envVar]?.split(ENV_VAR_PREFIX)[1],
```

Comment:
I don't think this is correct? See my previous comment — the feature flag name would need to be `envVar.substring(ENV_VAR_PREFIX.length)` or `envVar.split(ENV_VAR_PREFIX)[1]` (I prefer the substring version)

This code would be splitting the value of the environment variable.

Reply:
Oh yeah you're right this is backwards ><. And you're right, I'll use substrings, it is a bit clearer.

Reply:
https://github.com/inrupt/typescript-sdk-tools/pull/113/commits/559bfd0e3202f0cc2b77abb0fa8691fbd5ee9de6